src/main/java/de/undercouch/actson/JsonParser.java

```diff
@@ -399,14 +402,40 @@ private void parse(char nextChar) {
         event1 = stateToEvent();
       }
 
-      // Change the state.
       state = nextState;
+    } else if (nextState == XU) {
```

Comment:
Also currently no code in `performAction` modifies `this.currentValue`. So these sections that update `this.currentValue` seemed to me like a gray area :)